project/project.go

```diff
@@ -164,30 +166,89 @@ func (p *Project) load(file string, bytes []byte) error {
 
 	p.configVersion = version
 
-	for name, config := range serviceConfigs {
-		err := p.AddConfig(name, config)
+	for name, config := range volumeConfigs {
+		err := p.AddVolumeConfig(name, config)
 		if err != nil {
 			return err
 		}
 	}
 
-	for name, config := range volumeConfigs {
-		err := p.AddVolumeConfig(name, config)
+	for name, config := range networkConfigs {
+		err := p.AddNetworkConfig(name, config)
 		if err != nil {
 			return err
 		}
 	}
 
-	for name, config := range networkConfigs {
-		err := p.AddNetworkConfig(name, config)
+	for name, config := range serviceConfigs {
+		err := p.AddConfig(name, config)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Update network configuration a little bit
+	if p.isNetworkEnabled() {
+		for _, serviceName := range p.ServiceConfigs.Keys() {
+			serviceConfig, _ := p.ServiceConfigs.Get(serviceName)
+			if serviceConfig.NetworkMode != "" {
+				continue
+			}
+			if serviceConfig.Networks == nil || len(serviceConfig.Networks.Networks) == 0 {
+				// Add default as network
+				serviceConfig.Networks = &yaml.Networks{
+					Networks: []*yaml.Network{
+						{
+							Name: "default",
+						},
+					},
+				}
+			}
+			// Consolidate the name of the network
+			// FIXME(vdemeester) probably shouldn't be there, maybe move that to interface/factory
+			for _, network := range serviceConfig.Networks.Networks {
+				if net, ok := p.NetworkConfigs[network.Name]; ok {
+					if net.External.External {
+						network.RealName = network.Name
+						if net.External.Name != "" {
+							network.RealName = net.External.Name
+						}
+					} else {
+						network.RealName = p.Name + "_" + network.Name
+					}
+				}
+				// Ignoring if we don't find the network, it will be catched later
+			}
+		}
+	}
+
+	// FIXME(vdemeester) Not sure about this..
+	if p.context.NetworksFactory != nil {
+		networks, err := p.context.NetworksFactory.Create(p.Name, p.NetworkConfigs, p.ServiceConfigs, p.isNetworkEnabled())
 		if err != nil {
 			return err
 		}
+
+		p.networks = networks
 	}
 
 	return nil
 }
 
+func (p *Project) isNetworkEnabled() bool {
+	return p.configVersion == "2"
```

Comment:
If no networks are specified, it looks like there's a slight difference in behavior between v1 and v2.
- v1 will behave exactly the same as before this PR and make no network API calls
- v2 will create a create a `default` network

Is this correct?